### DISCUSSION TITLE
Impact of compression on scrubs

### DISCUSSION THREAD

**Discussion by aaronjwood** [openzfs/zfs#13139] [Q&A] [answered]
I'm trying to understand how compression is treated during a scrub. Let's say I have a dataset or zvol that's 1TB in size with a 50% compression (1.50x compressratio). Is the data first decompressed and then checked, effectively scrubbing the full 1TB of data, or is only 500GB of data scrubbed?

**Comment by aaronjwood** [comment, marked as answer]
What I'm trying to understand from my question is what's the order of the operations in the chain. I ended up finding the answer: `When a file is written, the data is compressed, encrypted, and the checksum is verified. Then, the data is deduplicated, if possible.`

**Reply by rincebrain** [reply]
Scrubs check the checksum of the data on disk, which is after compression/encryption. They aren't going to try decompressing or decrypting it.
